### ISSUE TITLE
"workspace move to output" creates ws on original output based on name instead of number

### ISSUE THREAD

**Issue by antoinemadec** [i3/i3#3745]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Let's say we work with 2 monitors, with only 2 empty workspaces.

When the ws is only a number:
```i3
workspace 1
move workspace to output right
```
- moves **ws 1** on right output
- creates **ws 11** on left output

When the ws is a number and a name:
```i3
workspace 1:foo
move workspace to output right
```
- moves **ws 1:foo** on right output
- creates **ws 1** on left output

## Expected Behavior

```move workspace to output``` should create a new workspace on the original output based on the existing workspaces *number*, not *name*.

## Reproduction Instructions

See **Current Behavior**

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.16.1-195-g90385edc (2019-07-21, branch "gaps-next") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16.1-195-g90385edc (2019-07-21, branch "gaps-next") (pid 940)
Loaded i3 config: /home/antoine/.config/i3/config (Last modified: Thu 25 Jul 2019 11:43:56 AM PDT, 1068 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
The default i3 config reproduces the bug.
</pre>

<pre>
Logfile URL:
</pre>

<pre>
- Linux Distribution & Version: Debian_GNULinux_10_buster
- Are you using a compositor (e.g., xcompmgr or compton): yes, compton
</pre>

**Comment by i3bot** [comment]
I don’t see a link to logs.i3wm.org. Did you follow https://i3wm.org/docs/debugging.html? (In case you actually provided a link to a logfile, please ignore me.)

**Comment by Airblader** [comment]
I get were you're coming from and in how I use workspaces I'd agree. But duplicate workspace numbers are not forbidden in i3, so I'm not sure this isn't actually an opionated change.

It also has the potential to break behavior for users. We should probably think about this one. 

For most users with predefined workspace names in bindings this is probably a non-issue either way.

**Comment by antoinemadec** [comment]
@Airblader , got it!

It is however surprising that,
if you only have 2 ws: **1:foo** and **2**,
the following command will jump into **1:foo**:
```
workspace 1
```
Meaning that both **1** and **1:foo** seem to describe the same ws for some i3 command.
However ```move workspace to output``` make the distinction between both.

I get that **1** and **1:foo** can coexist and that some people might find it useful.
But I think the commands should behave consistently by treating **1** and **1:foo** the same unless explicitly specify by the user.

More importantly, that default behavior makes the use of ```<mod>+<num>``` useless.

What do you think?

**Comment by Airblader** [comment]
> the following command will jump into 1:foo:

Yeah, I believe a few years back we changed that, it caused *a lot* of complaints from users and we reverted it back. Theoretically speaking, I think it's incorrect behavior and `workspace number 1` should be used instead.

My worry is that while inconsistent, changing the behavior could have the same effect. But thinking about it I feel like this case is different, and that in the majority of cases it probably would do what the user expects.
